presto-main/src/main/java/com/facebook/presto/operator/Driver.java

```diff
@@ -86,23 +86,30 @@
     private TaskSource currentTaskSource;
 
     private final AtomicReference<SettableFuture<?>> driverBlockedFuture = new AtomicReference<>();
-    private final AtomicBoolean initialized = new AtomicBoolean();
 
     private enum State
     {
         ALIVE, NEED_DESTRUCTION, DESTROYED
     }
 
-    public Driver(DriverContext driverContext, Operator firstOperator, Operator... otherOperators)
+    public static Driver createDriver(DriverContext driverContext, List<Operator> operators)
     {
-        this(requireNonNull(driverContext, "driverContext is null"),
+        Driver driver = new Driver(requireNonNull(driverContext, "driverContext is null"), requireNonNull(operators, "operators is null"));
```

Comment:
Move the `requireNonNull` checks to a separate line. Calling them inline looks messy.